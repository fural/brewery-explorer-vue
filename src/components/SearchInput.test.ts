import { fireEvent, render } from "@testing-library/vue";
import component from "./SearchInput.vue";

const defaultProps = {
  placeholder: "test-search",
  name: "test-search",
  modelValue: "test-model-value",
};

test("it should match snapshot", async () => {
  // Arrange
  const { container, getByRole } = render(component, {
    props: defaultProps,
  });

  // Assert
  expect(getByRole("textbox", { name: "test-search" })).toBeTruthy();
  expect(container).toMatchSnapshot();
});

test("it should emitt the input event", async () => {
  // Arrange
  const { emitted, getByRole } = render(component, {
    props: defaultProps,
  });

  const input = getByRole("textbox", { name: "test-search" });

  // Act
  await fireEvent.update(input, "testing str");

  // Assert
  //@ts-ignore
  expect(input.value).toBe("testing str");
  expect(emitted("update:modelValue")).toBeTruthy();
});
