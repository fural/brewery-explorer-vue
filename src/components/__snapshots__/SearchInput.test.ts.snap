// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it should match snapshot 1`] = `
<div>
  <label
    class="sm:w-3/5 relative"
  >
    <span
      class="sr-only"
    >
      test-search
    </span>
    <span
      class="absolute inset-y-0 left-0 flex items-center pl-2"
    >
      <svg
        class="h-5 w-5"
        fill="none"
        stroke="currentColor"
        strokeWidth="{1.5}"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </span>
    <input
      class="pl-9"
      name="test-search"
      placeholder="test-search"
      title="test-search"
      type="text"
    />
  </label>
</div>
`;
